bgpd/bgp_clist.c

```diff
@@ -337,18 +337,38 @@ static void community_list_entry_delete(struct community_list_master *cm,
 		community_list_delete(cm, list);
 }
 
+/*
+ * Replace community-list entry in the list. Note that entry is the new one
+ * and replace is one one being replaced.
+ */
+static void community_list_entry_replace(struct community_list *list,
+					 struct community_entry *replace,
+					 struct community_entry *entry)
+{
+	if (replace->next) {
+		entry->next = replace->next;
+		replace->next->prev = entry;
+	} else
+		list->tail = entry;
```

Comment:
Do we need to make sure  entry->next is set to NULL ?

Reply:
I'll add setting NULL for both entry->next and entry->prev (for the following block)

Reply:
just curious: Is this really needed? For me personally, does not sound the best place for manipulating next/prev here. Or I don't understand something here :)

Reply:
We're replacing an old community list entry with another one with the same sequence number.  How else would we set the next pointer of the previous entry (if it exists) or the prev pointer of the next entry (if it exists).  I'm certainly open to suggestions.